lib/src/utils/screen_loading/screen_loading_manager.dart

```diff
@@ -0,0 +1,220 @@
+import 'dart:developer';
+
+import 'package:flutter/foundation.dart';
+import 'package:instabug_flutter/instabug_flutter.dart';
+import 'package:instabug_flutter/src/utils/screen_loading/flags_config.dart';
+import 'package:instabug_flutter/src/utils/screen_loading/screen_loading_trace.dart';
+import 'package:instabug_flutter/src/utils/screen_loading/ui_trace.dart';
+import 'package:meta/meta.dart';
+
+class ScreenLoadingManager {
+  ScreenLoadingManager._();
+
+  static ScreenLoadingManager _instance = ScreenLoadingManager._();
+
+  static ScreenLoadingManager get instance => _instance;
+
+  /// Shorthand for [instance]
+  static ScreenLoadingManager get I => instance;
+  static const tag = "ScreenLoadingManager";
+  UiTrace? _currentUiTrace;
+  ScreenLoadingTrace? _currentScreenLoadingTrace;
+
+  /// @nodoc
+  @internal
+  ScreenLoadingTrace? get currentScreenLoadingTrace => _currentScreenLoadingTrace;
+
+  /// @nodoc
+  @internal
+  final List<ScreenLoadingTrace> prematurelyEndedTraces = [];
+
+  /// @nodoc
+  @internal
+  UiTrace? get currentUiTrace => _currentUiTrace;
+
+  @visibleForTesting
+  // ignore: use_setters_to_change_properties
+  static void setInstance(ScreenLoadingManager instance) {
+    _instance = instance;
+  }
+
+
+  /// @nodoc
+  @internal
+  void resetDidStartScreenLoading() {
+    // Allows starting a new screen loading capture trace in the same ui trace (without navigating out and in to the same screen)
+    _currentUiTrace?.didStartScreenLoading = false;
+    debugPrint('${APM.tag}: Resetting didStartScreenLoading — setting didStartScreenLoading: ${_currentUiTrace?.didStartScreenLoading}');
+  }
+
+  /// @nodoc
+  @internal
+  void resetDidReportScreenLoading() {
+    // Allows reporting a new screen loading capture trace in the same ui trace even if one was reported before by resetting the flag which is used for checking.
+    _currentUiTrace?.didReportScreenLoading = false;
+    debugPrint('${APM.tag}: Resetting didReportScreenLoading — setting didReportScreenLoading: ${_currentUiTrace?.didReportScreenLoading}');
+  }
+
+  @internal
+  Future<void> startUiTrace(String screenName) async {
+    resetDidStartScreenLoading();
+    final isApmEnabled = await FlagsConfig.Apm.isEnabled();
+    if (!isApmEnabled) {
+      log("Unable to start Ui Trace, as ${FlagsConfig.Apm.name} feature is disabled.",
+          name: APM.tag);
+      return;
+    }
+    final microTimeStamp = DateTime.now().microsecondsSinceEpoch;
+    final uiTraceId = DateTime.now().millisecondsSinceEpoch;
```

Comment:
@a7medev Since native SDK wanted to prevent wasting space as well as storing it redundantly here, they've suggested dividing it by 1000 and casting it as int before sending it. Therefore I'll keep it as is for now.